voyager-transitions/src/commonMain/kotlin/cafe/adriel/voyager/transitions/ScreenTransition.kt

```diff
@@ -73,17 +102,51 @@ public fun ScreenTransition(
             enter togetherWith exit
         },
         modifier = modifier,
+        disposeScreenAfterTransitionEnd = disposeScreenAfterTransitionEnd,
+        content = content
+    )
+}
+
+@Composable
+public fun ScreenTransition(
+    navigator: Navigator,
+    transition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
+    modifier: Modifier = Modifier,
+    content: ScreenTransitionContent = { it.Content() }
+) {
+    ScreenTransition(
+        navigator = navigator,
+        transition = transition,
+        modifier = modifier,
+        disposeScreenAfterTransitionEnd = false,
         content = content
     )
 }
 
+@ExperimentalVoyagerApi
+@OptIn(ExperimentalAnimationApi::class)
 @Composable
 public fun ScreenTransition(
     navigator: Navigator,
     transition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
     modifier: Modifier = Modifier,
+    disposeScreenAfterTransitionEnd: Boolean = false,
     content: ScreenTransitionContent = { it.Content() }
 ) {
+    val screenCandidatesToDispose = rememberSaveable(saver = screenCandidatesToDisposeSaver()) {
+        mutableStateOf(emptySet())
+    }
+
+    val currentScreens = navigator.items
+
+    DisposableEffect(currentScreens) {
```

Comment:
suggest to wrap 142-148 lines in if condition `if (disposeScreenAfterTransitionEnd) { ... }`